### ISSUE TITLE
Non-jquery upload script to get things such as upload progress

### ISSUE THREAD

**Issue by AndreasGalster** [cloudinary/cloudinary_js#83]
It seems that there's no vanilla-js version of a upload script yet. I am trying to write a Polymer Webcomponent for cloudinary uploads and am currently stuck at getting the progress of the individual upload. Is jquery.fileupload.js the only available option? Does it need to be ported?

**Comment by taragano** [comment]
Hi @AndreasGalster,
I apologize for the response delay.

Have you checked our [Angular SDK](https://github.com/cloudinary/cloudinary_angular)?
You'll be able to upload with Angular too if that's what you prefer. It includes a sample project that you can check out.

**Comment by zenbrent** [comment]
I'd like to put in a vote for a version of the library that doesn't have any dependencies.
I don't use angular or jQuery. Many of the browser inconsistencies that made jQuery useful are irrelevant in modern browsers, and more importantly, there are plenty of polyfills and shims and shams to make standard browser APIs work as expected in older browsers. I don't think it's bad to use jQuery, but I don't need it for my own code and I avoid libs that have it as a dependancy.
As far as the Angular version... I like React. At my last job I used Backbone with different supporting frameworks. At my current job, we've been using Mithril. I may use Angular sometime, but the Angular SDK doesn't help now or with anything I've ever done.
It would be awesome if I could use all of Cloudinary's features with no dependancies other than modern, standards-compliant browser APIs (which can be polyfilled for older browsers anyway.)

**Comment by reintroducing** [comment]
I was going to post a new issue about this but it seems like there is already discussion here.  I'm building a React app and using Cloudinary for images (display and subsequently/hopefully uploading). I have no interest in including jQuery in my completely React/Redux app just to handle uploading images to Cloudinary.

What is the stance of Cloudinary on image uploading for non-jQuery javascript objects?  I think it's safe to say React is staying around now and a solution needs to be available for this.  

In short, what is the official Cloudinary team's recommendation on how to upload images to Cloudinary in a non jQuery JS application that is framework agnostic?

**Comment by AndreasGalster** [comment]
Any updates on this? We're currently stuck and are unable to implement cloudinary because we're not sure how to implement cloudinary server side (docs aren't very helpful, btw they're also very out of date!) and client side doesn't look like an option for us as well.

**Comment by taragano** [comment]
I'm sorry for my response delay everyone.
It's on our road-map to add support to pure javascript uploads in the future. There's no ETA for this at the moment.

**Comment by Thebigbignooby** [comment]
Where could I find documentation and how to upload images to cloudinary without the SDKs ? I'm not talking about a vanilla JS sdk, I'm oging to implement hte uploading myself, but I don't know what kind of object expect.

**Comment by taragano** [comment]
Here's a basic example code, which demonstrates unsigned uploads using pure javascript without any SDK:
http://jsfiddle.net/tocker/fpoca4c8/

Let me know if this helps?

**Comment by roeeba** [comment]
Closing this issue due to the time elapsed. Please feel free to either re-open the issue, contact our support at http://support.cloudinary.com or create a new ticket if you have any additional issues.
